Updated string converters

joseph-wortmann opened the pull request:
- Updated the Segment class to use the super string converter by default
- Added a string converter to the Message class to address missing ‘\r’ on the final segment of the message
- added the .vscode settings directory to .gitignore

johnpaulett commented:
Thanks for the contribution.  

Not sure why GH Actions didn't test this PR -- I'll need to look at that.  But from a quick look at the code, is the [test suite](https://python-hl7.readthedocs.io/en/latest/contribute.html#testing) passing? Modifying Container.__str__ is likely to add a trailing separator to the Fields & Repetitions, not just to the Segment

tox may not fully work on your machine if all the python versions don't exist, but that's ok -- as long as one of the python version passes.

joseph-wortmann commented:
I didn't modify Containter.__str__. I changed the Segment.__str__ to use the super version when not a 'MSH' segment and I added a Message.__str__ to add the final, trailing '\r' that was missing on the final segment.

Some of the tests are failing asserts due to the addition of the final '\r'. Should I update those tests to passing?

joseph-wortmann commented:
@johnpaulett I went ahead and updated the tests to passing by making sure the trailing '\r' was in the asserted strings.

johnpaulett commented:
FYI, the doctests were broken too, fixed in 2416741
